fineract-core/src/main/java/org/apache/fineract/batch/service/BatchApiServiceImpl.java

```diff
@@ -116,7 +116,7 @@ public List<BatchResponse> handleBatchRequestsWithoutEnclosingTransaction(final
      */
     @Override
     public List<BatchResponse> handleBatchRequestsWithEnclosingTransaction(final List<BatchRequest> requestList, final UriInfo uriInfo) {
-        return callInTransaction(Function.identity()::apply, () -> handleBatchRequests(true, requestList, uriInfo));
+        return callInTransaction(Function.identity()::apply, () -> handleBatchRequests(true, requestList, uriInfo), true);
```

Comment:
Could you please explain why this was needed? The enclosing transaction was put both cases but
withEnclosingTransaction around the whole batch and withoutEnclosingTransaction around each separate execution.
If you change this then there is a use-case with withoutEnclosingTransaction which might not work: when the parent request fails then all the child requests should also fail, in method parentRequestFailedRecursive we should mark the child transactions to be rolled back. Is this working fine after your change?
Additionally this story is about to handle both with- and without enclosing transaction cases - to save the failed command - and I do not see how this helps in case of withEnclosing case. 
With this change, as I see, the problem is not solved, but postponed.

Reply:
I did this so that when calling the withoutEnclosingTransaction method, the 'sameTransaction' flag in the executeCommand method of the SynchronousCommandProcessingService was false and then saveInitialNewTransaction was called, because this guarantees that unsuccessful sub-requests are recorded in the trail audit logs. As for the use-case `when the parent request fails then all the child requests should also fail, in method parentRequestFailedRecursive we should mark the child transactions to be rolled back`, after my changes nothing seems to be broken. I tested it again today.

Reply:
You are right that enclosingTransaction flag should be correctly set for 'withoutEnclosingTransaction' case to false. This was a bug and will be fixed with the 4-eye principle PR 
https://github.com/apache/fineract/pull/3649
as maker-checker concept was also not working in this case.
Unfortunately this does not solve the batch api problem of this story.